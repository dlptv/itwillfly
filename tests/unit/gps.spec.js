const SerialPort = require('serialport/test');

const { Binding: MockBinding } = SerialPort;

// needed to mock SerialPort
// eslint-disable-next-line global-require
jest.mock('serialport', () => require('serialport/test'));

const GPS = require('../../src/gps');

const portPath = '/dev/ttyTest';

test('GPS module reads RMC data from port', (done) => {
  MockBinding.createPort(portPath, { echo: false, record: false });
  const port = new SerialPort(portPath, { baudRate: 115200 });

  GPS.createListener(port, (data) => {
    const expectedData = {
      time: new Date('2017-12-12T18:23:25.921Z'), lat: 59.05983333333333, lon: 5.933033333333333, speed: 35.9288, track: 127.8,
    };
    expect(data).toMatchObject(expectedData);
    done();
  });

  port.on('open', () => {
    port.binding.emitData(Buffer.from('$GPRMC,182325.921,A,5903.590,N,00555.982,E,019.4,127.8,121217,000.0,W*7C\n'));
  });
});
