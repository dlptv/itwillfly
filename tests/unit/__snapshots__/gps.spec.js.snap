// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GPS module reads RMC data from port 1`] = `
Object {
  "faa": null,
  "lat": 59.05983333333333,
  "lon": 5.933033333333333,
  "raw": "$GPRMC,182325.921,A,5903.590,N,00555.982,E,019.4,127.8,121217,000.0,W*7C",
  "speed": 35.9288,
  "status": "active",
  "time": 2017-12-12T18:23:25.921Z,
  "track": 127.8,
  "type": "RMC",
  "valid": true,
  "variation": -0,
}
`;
